lib/model/message.dart

```diff
@@ -60,14 +60,17 @@ class MessageStoreImpl with MessageStore {
     }
   }
 
-  void dispose() {
-    // When a MessageListView is disposed, it removes itself from the Set
-    // `MessageStoreImpl._messageListViews`. Instead of iterating on that Set,
-    // iterate on a copy, to avoid concurrent modifications.
-    for (final view in _messageListViews.toList()) {
-      view.dispose();
-    }
-  }
+  // No `dispose` method, because there's nothing for it to do.
+  // Not disposing the [MessageListView]s here, because they are owned by
+  // (i.e., they get [dispose]d by) the [_MessageListState], including in the
+  // case where the [PerAccountStore] is replaced.
+  //
+  // TODO: Add assertions that the [MessageListView]s are indeed disposed, by
+  //   first ensuring that [PerAccountStore] is only disposed after those with
+  //   references to it are disposed, then reinstating this `dispose` method.
+  //   Discussion:
+  //     https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/.60MentionAutocompleteView.2Edispose.60/near/2083074
```

Comment:
nit: let's have this point here:
https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/MessageListView.20lifecycle/near/2086893

since that's where the discussion relevant to this particular TODO begins.